pkg/config/templateComponent.go

```diff
@@ -62,13 +65,33 @@ type TemplateData struct {
 // we add new components.
 type TemplateComponent struct {
 	Name        string             `yaml:"name"`
+	CollName    string             `yaml:"collName"`
 	Kind        string             `yaml:"kind"`
 	Summary     string             `yaml:"summary,omitempty"`
 	Description string             `yaml:"description,omitempty"`
 	Ports       []TemplatePort     `yaml:"ports,omitempty"`
 	Properties  []TemplateProperty `yaml:"properties,omitempty"`
 	Templates   []TemplateData     `yaml:"templates,omitempty"`
 	User        map[string]any     `yaml:"user,omitempty"`
+	hpsf        *hpsf.Component    // the component from the hpsf document
+	connections []*hpsf.Connection `yaml:"connections,omitempty"`
+}
+
+// we're making a copy here to make sure it doesn't get modified
+func (t *TemplateComponent) SetHPSF(c hpsf.Component) {
+	t.hpsf = &c
+}
+
+// HProp is a template helper that gets a map of all properties specified in the hpsf document.
+func (t *TemplateComponent) HProps() map[string]any {
```

Comment:
I'm not sure HProps is clear enough for what it is but also HPSFProps maybe feels cumbersome?

Reply:
We definitely want these names to be short because they're in templates that can't easily be broken into multiple lines; I chose this because it's like Props but different. Let's live with it for a little while; it's not hard to change if we find it confusing.